jdaviz/app.py

```diff
@@ -542,6 +543,33 @@ def add_data_to_viewer(self, viewer_reference, data_label,
                 f"of:\n\t" + f"\n\t".join([
                     data_item['name'] for data_item in self.state.data_items]))
 
+    def _set_plot_axes_labels(self, data, viewer_id):
+        """
+        Sets the plot axes labels to be the units of the data to be loaded.
+
+        Parameters
+        ----------
+        data : `~Spectrum1D`
+            Data from ``DataCollection`` that will have its units as the plot
+            label axes.
+        viewer_id : str
+            The UUID associated with the desired viewer item.
+        """
+        viewer = self._viewer_by_id(viewer_id)
+
+        # Get the units of the data to be loaded.
+        spectral_axis_unit_type = data.spectral_axis.unit.physical_type.title()
+        flux_unit_type = data.flux.unit.physical_type.title()
+
+        if spectral_axis_unit_type.lower() == "length".lower():
+            spectral_axis_unit_type = "Wavelength"
+
```

Comment:
I think energy and frequency make sense as-is, but one more spectral axis item needs special-casing:
```suggestion
        if data.spectral_axis.unit.is_equivalent(u.pixel):
            spectral_axis_unit_type = "pixel"
```